src/App.js

```diff
@@ -1,19 +1,32 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function useSemiPersistentState(key, initialState) {
+  const [value, setValue] = useState(() => {
+    const savedValue = JSON.parse(localStorage.getItem(key));
+    return savedValue !== null ? savedValue : initialState;
+  });
+
+  useEffect(() => {
+    localStorage.setItem(key, JSON.stringify(value));
+  }, [key, value]);
+
+  return [value, setValue];
+}
```

Comment:
This is great and I love seeing a ternary show up! My only notes are purely stylistic/opinion based. If what you wrote reads well and makes sense to you that's awesome, I just generally take a semantic approach to naming. For example, `value`, `setValue`, and `key` aren't super descriptive and could refer to a number of things. I would name the state variable and setter after what they refer to, something like `todoList` and `setTodoList`. Then `key` could be something like `savedTodoList`. Again, this is just my opinion as it works totally fine either way. Below is another approach using a more semantic naming convention:

const useSemiPersistentState = () => {
  const [todoList, setTodoList] = useState(
    JSON.parse(localStorage.getItem('savedTodoList')) || []
  );

  useEffect(() => {
    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
  }, [todoList]);

  return [todoList, setTodoList];
};